### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Will push at noon if no one reviews

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore.Design/Design/DbContextOperations.cs [hunk 1]
```diff
@@ -136,17 +136,17 @@ public virtual Type GetContextType([CanBeNull] string name)
             {
                 contexts.Add(
                     context,
-                    FindContextFactory(context) ?? (Func<DbContext>)(() =>
-                    {
-                        try
+                    FindContextFactory(context) ?? (() =>
                         {
```
Comment (bricelam) [inline src/Microsoft.EntityFrameworkCore.Design/Design/DbContextOperations.cs]: Does this conflict with the VS default auto-format indentation? I can't stand when they flip back and forth every commit.

File: src/Microsoft.EntityFrameworkCore.InMemory/Query/ExpressionVisitors/Internal/InMemoryEntityQueryableExpressionVisitorFactory.cs [hunk 2]
```diff
@@ -39,11 +39,11 @@ public class InMemoryEntityQueryableExpressionVisitorFactory : IEntityQueryableE
...
         /// </summary>
         public virtual ExpressionVisitor Create(
-            EntityQueryModelVisitor queryModelVisitor, IQuerySource querySource)
-            => new InMemoryEntityQueryableExpressionVisitor(
-                _model,
-                _materializerFactory,
-                Check.NotNull(queryModelVisitor, nameof(queryModelVisitor)),
-                querySource);
+            EntityQueryModelVisitor queryModelVisitor, IQuerySource querySource
+        ) => new InMemoryEntityQueryableExpressionVisitor(
```
Comment (bricelam) [inline src/Microsoft.EntityFrameworkCore.InMemory/Query/ExpressionVisitors/Internal/InMemoryEntityQueryableExpressionVisitorFactory.cs]: That's an odd place to put a parenthesis

File: src/Microsoft.EntityFrameworkCore.Relational.Specification.Tests/NullSemanticsQueryTestBase.cs [hunk 3]
```diff
@@ -356,7 +354,7 @@ public virtual void Where_equal_nullable_with_null_value_parameter()
             string prm = null;
 
             AssertQuery<NullSemanticsEntity1>(es => es.Where(e =>
-                e.NullableStringA == prm));
+                    e.NullableStringA == prm));
```
Comment (bricelam) [inline src/Microsoft.EntityFrameworkCore.Relational.Specification.Tests/NullSemanticsQueryTestBase.cs]: I don't understand the reasoning for the double indent here. It's only nice if there's something after it.

File: src/Microsoft.EntityFrameworkCore.InMemory/Query/ExpressionVisitors/Internal/InMemoryEntityQueryableExpressionVisitorFactory.cs (hunk 2 above)
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore.InMemory/Query/ExpressionVisitors/Internal/InMemoryEntityQueryableExpressionVisitorFactory.cs]: Without the parenthesis resharper adds an extra ident to parameters and body, I think it's a bug

File: src/Microsoft.EntityFrameworkCore.Relational.Specification.Tests/NullSemanticsQueryTestBase.cs (hunk 3 above)
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore.Relational.Specification.Tests/NullSemanticsQueryTestBase.cs]: Probably parenthesis and lambda rules overlapping

File: src/Microsoft.EntityFrameworkCore.Design/Design/DbContextOperations.cs (hunk 1 above)
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore.Design/Design/DbContextOperations.cs]: VS formats it the same way. At least on my machine.

**Comment by ajcvickers** [comment]
Can you summarize what issues with the current formatting rules you are attempting to fix? (I wasn't aware of anything terrible with what we had.)

File: src/Microsoft.EntityFrameworkCore.InMemory/Query/ExpressionVisitors/Internal/InMemoryEntityQueryableExpressionVisitorFactory.cs (hunk 2 above)
Comment (AndriySvyryd) [inline src/Microsoft.EntityFrameworkCore.InMemory/Query/ExpressionVisitors/Internal/InMemoryEntityQueryableExpressionVisitorFactory.cs]: This wasn't caused by a rule change, but (probably) a bug in the current version. By default it would format it like this:
```C#
public virtual ExpressionVisitor Create(
        EntityQueryModelVisitor queryModelVisitor, IQuerySource querySource)
    => new InMemoryEntityQueryableExpressionVisitor(
        _model,
        _materializerFactory,
        Check.NotNull(queryModelVisitor, nameof(queryModelVisitor)),
        querySource);
```
Ideally it should be:
```C#
public virtual ExpressionVisitor Create(
    EntityQueryModelVisitor queryModelVisitor, IQuerySource querySource)
    => new InMemoryEntityQueryableExpressionVisitor(
        _model,
        _materializerFactory,
        Check.NotNull(queryModelVisitor, nameof(queryModelVisitor)),
        querySource);
```
Which is the closesest to statement body formatting:
```C#
public virtual ExpressionVisitor Create(
    EntityQueryModelVisitor queryModelVisitor, IQuerySource querySource)
{
    return new InMemoryEntityQueryableExpressionVisitor(
        _model,
        _materializerFactory,
        Check.NotNull(queryModelVisitor, nameof(queryModelVisitor)),
        querySource);
}
```
But I couldn't find any rule to achieve this.
So the workaround is to move the parenthesis:
```C#
public virtual ExpressionVisitor Create(
    EntityQueryModelVisitor queryModelVisitor, IQuerySource querySource
) => new InMemoryEntityQueryableExpressionVisitor(
    _model,
    _materializerFactory,
    Check.NotNull(queryModelVisitor, nameof(queryModelVisitor)),
    querySource);
```

**Comment by AndriySvyryd** [comment]
The above is the only issue I've seen

File: src/Microsoft.EntityFrameworkCore.InMemory/Query/ExpressionVisitors/Internal/InMemoryEntityQueryableExpressionVisitorFactory.cs (hunk 2 above)
Comment (ajcvickers) [inline src/Microsoft.EntityFrameworkCore.InMemory/Query/ExpressionVisitors/Internal/InMemoryEntityQueryableExpressionVisitorFactory.cs]: To me the "default" one looks better than the "workaround" one.
